app/uk/gov/hmrc/taxenrolmentassignmentfrontend/controllers/MultipleAccountsCheckController.scala

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.taxenrolmentassignmentfrontend.controllers
+
+import play.api.i18n.I18nSupport
+import play.api.mvc._
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+import uk.gov.hmrc.taxenrolmentassignmentfrontend.config.AppConfig
+import uk.gov.hmrc.taxenrolmentassignmentfrontend.connectors.EACDConnector
+import uk.gov.hmrc.taxenrolmentassignmentfrontend.controllers.auth.AuthAction
+import uk.gov.hmrc.taxenrolmentassignmentfrontend.errors.UnexpectedResponseFromEACD
+import uk.gov.hmrc.taxenrolmentassignmentfrontend.views.html.UnderConstructionView
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.ExecutionContext
+
+@Singleton
+class MultipleAccountsCheckController @Inject()(
+  authAction: AuthAction,
+  mcc: MessagesControllerComponents,
+  eacdConnector: EACDConnector,
+  underConstructionView: UnderConstructionView
+)(implicit ec: ExecutionContext,
+  implicit val config: AppConfig)
+    extends FrontendController(mcc)
+      with I18nSupport {
+
+
+  def multipleAccountsCheck(redirectUrl: String): Action[AnyContent] = authAction.async {
+    implicit request =>
+        eacdConnector.getUsersWithPTEnrolment(request.userDetails.nino).value.map {
+          case Right(Some(credsWithPTE)) if credsWithPTE.principalUserIds.contains(request.userDetails.credId) =>
+            Redirect(redirectUrl)
```

Comment:
Is this a valid scenario? This is essentially checking if the current user has a PT enrolment, but we already do that in the auth action, so shouldn't the user be redirected from that, or just use the 'hasPTEnrolment' boolean?

Reply:
Technically it shouldn't happen but if a users is signing in on multiple devices there is a possibility it could. If this check isn't put in then the user will be presented with a page asking them to sign in to the account they are already signed in with.

Reply:
I'm not following, if the user is signed in on multiple devices but with the same account, it shouldn't change the behaviour right? It would still recognise whether they have a PT enrolment or not with the auth check, or are you talking about a different scenario?

Reply:
You need to remember session are not user specific. If you are signed into two or more devices the session will be different. If in one you then enrolled for PT, that session would be updated to include the PT enrolment but I don't believe every session the user has will be.

Reply:
Oh you mean the user completes the journey on one device! I assumed you meant they had no enrolment on either device and had only just been redirected. 

Hmm this is very niche, essentially the question is just how long does it take for an enrolment to be recognised by auth right vs checking EACD? I'm not actually sure about the answer since this check seems like it would be better placed within the auth action. It would then redirect regardless of which page the user is on whereas here we only check once on this route.

Regardless, this is fine. We're/I'm probably overthinking it.